server/src/updaters/relationship-updaters.js

```diff
@@ -0,0 +1,202 @@
+// @flow
+
+import type { Viewer } from '../session/viewer';
+
+import invariant from 'invariant';
+
+import {
+  type RelationshipStatus,
+  relationshipStatuses,
+  type RelationshipRequest,
+} from 'lib/types/relationship-types';
+import { ServerError } from 'lib/utils/errors';
+
+import { dbQuery, SQL } from '../database';
+
+async function updateRelationship(
+  viewer: Viewer,
+  request: RelationshipRequest,
+) {
+  const { userID, status } = request;
+
+  if (viewer.id === userID) {
+    throw new ServerError('invalid_user');
+  }
+
+  const userQuery = SQL`SELECT 1 FROM users WHERE id = ${userID} LIMIT 1`;
+  const [users] = await dbQuery(userQuery);
+
+  if (!users.length) {
+    throw new ServerError('user_does_not_exist');
+  }
+
+  // friend request
+  if (status === relationshipStatuses.PENDING_FRIEND) {
+    const [relationships] = await fetchRelationshipQuery(viewer.id, userID);
+
+    if (!relationships.length) {
+      await createRelationshipQuery(viewer.id, userID, status);
+    } else {
+      const [actorRelationship] = getRelationshipsForActorUser(
+        relationships,
+        viewer.id,
+      );
+      if (actorRelationship) {
+        const message = actorFriendshipErrorMap[actorRelationship.status];
+        throw new ServerError(message);
+      }
+
+      const [receiverRelationship] = getRelationshipsForReceiverUser(
+        relationships,
+        viewer.id,
+      );
+      if (receiverRelationship) {
+        const message = receiverFriendshipErrorMap[receiverRelationship.status];
```

Comment:
If you send a request to somebody who has already sent you a request, I think we should treat this like the receiver accepting the friend request

Reply:
I did it this way because I'm handling acceptance when `FRIEND` status is provided. It follows requester intention I think. Do you think we should accept a friend request in both scenarios or we should get rid of `FRIEND` and accept only 'PENDING_FRIEND' status as the input of this function to accept a friend request?

Reply:
Good call, you're right that there's no difference. Let's make it so `update_relationships` can only accept the following:

- `0 - KNOW_OF` to "unfriend" aka "delete relationship"
- `2 - FRIEND` which handles requesting a new friend and accepting a friend request
- `3 - BLOCKED` which handles blocking a user